### ISSUE TITLE
Composer global configuration

### ISSUE THREAD

**Issue by tazorax** [renovatebot/renovate#11980]
### What would you like Renovate to be able to do?

Add a capability to specify the content of `$COMPOSER_HOME/config.json` file for Composer.

It will allow me to specify 2 configurations :
- private registries (Satis self-hosted) globally (and not in `composer.json`'s project)
- disable `secure-http` option globally

### If you have any ideas on how this should be implemented, please tell us here.

Maybe like it is done for `.npmrc` file and the `npmrc` option ?

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by tooppaaa** [comment]
This is definitely a feature I need as well !! Thanks a lot @tazorax  !

**Comment by rarkins** [comment]
Yes, please reuse the `detectGlobalManagerConfig` I added very recently to detect global `.npmrc` files too. Hopefully it works smoothly the same way

**Comment by rarkins** [comment]
Note: if reading the global config requires new *options* to be created (disable secure http for example - I don't know exactly what that means) then it should be a separate PR than the one which simply enables reading config from a global composer file at startup.

**Comment by tazorax** [comment]
Finally, I think I will add a list to configuration to set using `composer config -g` command

**Comment by Nikdro** [comment]
Hi there! 

Are there any news on this? Or is there a possible workaround available?   
We just started using renovate and really need this for our setup. 

Thanks

**Comment by rarkins** [comment]
@Nikdro how are you running Renovate and what exactly is not working for you without this feature?

**Comment by swichers** [comment]
@rarkins The first comment had the use case:

> private registries (Satis self-hosted) globally (**and not in composer.json's project**)

The current documentation (https://docs.renovatebot.com/php/#private-packages ) assumes you will be adding the private packagist configuration to the project's composer.json. That sort of design pattern isn't common with companies that manage a large number of private projects. ex: We have nearly 1000 private repositories.

It's a larger maintenance burden for us to maintain the packagist across these `composer.json` files vs configuring it at the global level. When we need to add additional repository configuration, we'd have to update the 1000~ private repositories.

If these commands are run once on a machine:

```sh
composer config --global repo.<packagist-name> composer '<packagist url>'
composer config --global --auth 'http-basic.<domain>' '<username>' '<token>'
```

The configuration applies for every repository.

Renovate isn't usable for us as-is because we cannot specify the global repo configuration.

**Comment by Nikdro** [comment]
@viceice What's the state here? I would love to have composer's global manager config also applied here, like it does for npm.

I'd also like to help implementing and/or testing that feature. But I've no idea of how and where it should be implemented.

**Comment by rarkins** [comment]
You can create a PR if you wish. Please do not @ maintainers like the above when the only reason is to get attention.

**Comment by zogot** [comment]
I see 2 comments marked as resolved. This is true for us as an Organization too, similar to @swichers with more than 400 repositories.

Any progress on this or a workaround?
